auv_control/auv_control/scripts/reference_pose_publisher.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python3
+
+import rospy
+from nav_msgs.msg import Odometry
+from geometry_msgs.msg import Pose, Twist
+from auv_msgs.srv import SetDepth, SetDepthRequest, SetDepthResponse
+from tf.transformations import quaternion_from_euler, euler_from_quaternion
+from auv_common_lib.control.enable_state import ControlEnableHandler
+
+
+class DepthControllerNode:
+    def __init__(self):
+        rospy.init_node("reference_pose_publisher_node")
+
+        # Initialize subscribers
+        self.odometry_sub = rospy.Subscriber(
+            "/taluy/odometry", Odometry, self.odometry_callback
+        )
+        self.set_depth_service = rospy.Service(
+            "/taluy/set_depth", SetDepth, self.target_depth_handler
+        )
+        self.cmd_vel_sub = rospy.Subscriber(
+            "/taluy/cmd_vel", Twist, self.cmd_vel_callback
+        )
+
+        # Initialize publisher
+        self.cmd_pose_pub = rospy.Publisher("/taluy/cmd_pose", Pose, queue_size=10)
+
+        self.control_enable_handler = ControlEnableHandler(1.0)
+
+        # Initialize internal state
+        self.target_depth = 0.0
+        self.target_heading = 0.0
+        self.last_cmd_time = rospy.Time.now()
+
+        # Parameters
+        self.update_rate = rospy.get_param("~update_rate", 10)
+
+        rospy.Timer(rospy.Duration(1.0 / self.update_rate), self.control_loop)
```

Comment:
I remember this implementation has its own problems when ROS is being shutdown. It was something that we encountered during the competition season, maybe @batuhanozer remembers

But the good practice is to simply remove the rospy.spin() in the run function and create the loop there using 

rate = ...
while not rospy.is_shutdown()
   ...
   rate.sleep()